### PR TITLE
[ISSUE #8310] complete server addressing  in plugin way

### PR DESCRIPTION
**Pull request by MRGUOKING**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

for #8310, complete server addrssing in plugin way

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: core/src/main/java/com/alibaba/nacos/core/cluster/ServerMemberManager.java [hunk 1]
```diff
@@ -58,27 +65,12 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Optional;
 import java.util.Set;
 import java.util.concurrent.ConcurrentSkipListMap;
 
-/**
```
Comment (brotherlu-xcq) [inline core/src/main/java/com/alibaba/nacos/core/cluster/ServerMemberManager.java:64]: why remove this javadoc? If it's outdated, please give the latest.

File: core/src/main/java/com/alibaba/nacos/core/cluster/ServerMemberManager.java [hunk 2]
```diff
@@ -93,12 +85,36 @@ public class ServerMemberManager implements ApplicationListener<WebServerInitial
     
     private static final String MEMBER_CHANGE_EVENT_QUEUE_SIZE_PROPERTY = "nacos.member-change-event.queue.size";
     
-    private static final int DEFAULT_MEMBER_CHANGE_EVENT_QUEUE_SIZE = 128;
```
Comment (brotherlu-xcq) [inline core/src/main/java/com/alibaba/nacos/core/cluster/ServerMemberManager.java:96]: why change the queue size?

File: core/src/main/java/com/alibaba/nacos/core/cluster/ServerMemberManager.java [hunk 3]
```diff
@@ -449,7 +546,7 @@ public void onApplicationEvent(WebServerInitializedEvent event) {
         String serverNamespace = event.getApplicationContext().getServerNamespace();
         if (SPRING_MANAGEMENT_CONTEXT_NAMESPACE.equals(serverNamespace)) {
             // ignore
-            // fix#issue https://github.com/alibaba/nacos/issues/7230
```
Comment (brotherlu-xcq) [inline core/src/main/java/com/alibaba/nacos/core/cluster/ServerMemberManager.java:452]: why change the comment?

**Review by brotherlu-xcq — CHANGES_REQUESTED** [review]

**Review by T6252M — APPROVED** [review]

**Review by brotherlu-xcq — APPROVED** [review]
LGTM

**Pull request merged**